src/bp/nlu-core/typings.ts

```diff
@@ -69,6 +69,13 @@ export type ListEntity = Readonly<{
   sensitive: boolean
 }>
 
+export type ListEntityWithCache = ListEntity & {
```

Comment:
Why do we need `ListEntityWithCache` and `ColdListEntityModel`  .. both are different, I agree but could we get rid of one ?

Reply:
`ListEntityWithCache` should actually go in the training-pipeline... 

My first implementation looked like:

```
export type TrainInput {
   ...
   list_entities: ListEntity[],
   list_entities_caches: {entityName: string, cache: EntityCacheDump}[]
   ...
}
```

But later in the pipeline, I had to do some sort of a zip to associate entities with correct cache.

Instead of making the training pipeline algorithm more complex, I decided to make the typing more complex because :

"Data dominates. If you've chosen the right data structures and organized things well, the algorithms will almost always be self-evident.".

What do you think? Maybe The type `ListEntityWithCache` just doesnt need a name?